mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -241,3 +239,189 @@ def delete_registered_model_tag(self, name, key):
+
+    def delete_model_version(self, name, version):
+        """
+        Delete model version in backend.
+
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :return: None
+        """
+        req_body = message_to_json(DeleteModelVersionRequest(name=name, version=str(version)))
+        self._call_endpoint(DeleteModelVersionRequest, req_body)
+
+    def get_model_version(self, name, version):
+        """
+        Get the model version instance by name and version.
+
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :return: A single :py:class:`mlflow.entities.model_registry.ModelVersion` object.
+        """
+        req_body = message_to_json(GetModelVersionRequest(name=name, version=str(version)))
+        response_proto = self._call_endpoint(GetModelVersionRequest, req_body)
+        return model_version_from_uc_proto(response_proto.model_version)
+
+    def get_model_version_download_uri(self, name, version):
+        """
+        Get the download location in Model Registry for this model version.
+        NOTE: For first version of Model Registry, since the models are not copied over to another
+              location, download URI points to input source path.
+
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :return: A single URI location that allows reads for downloading.
+        """
+        req_body = message_to_json(
+            GetModelVersionDownloadUriRequest(name=name, version=str(version))
+        )
+        response_proto = self._call_endpoint(GetModelVersionDownloadUriRequest, req_body)
+        return response_proto.artifact_uri
+
+    def search_model_versions(self, filter_string):
+        """
+        Search for model versions in backend that satisfy the filter criteria.
+
+        :param filter_string: A filter string expression. Currently supports a single filter
+                              condition either name of model like ``name = 'model_name'`` or
+                              ``run_id = '...'``.
+        :return: PagedList of :py:class:`mlflow.entities.model_registry.ModelVersion`
+                 objects.
+        """
+        req_body = message_to_json(SearchModelVersionsRequest(filter=filter_string))
+        response_proto = self._call_endpoint(SearchModelVersionsRequest, req_body)
+        model_versions = [model_version_from_uc_proto(mvd) for mvd in response_proto.model_versions]
+        return PagedList(model_versions, response_proto.next_page_token)
+
+    def set_model_version_tag(self, name, version, tag):
+        """
+        Set a tag for the model version.
+
+        :param name: Registered model name.
+        :param version: Registered model version.
+        :param tag: :py:class:`mlflow.entities.model_registry.ModelVersionTag` instance to log.
+        :return: None
```

Comment:
```suggestion
```

Can we remove this because this function always fails?

Reply:
Will do here and in the other methods that throw